src/components/AgreementSections/LinesList.js

```diff
@@ -30,12 +30,23 @@ export default class LinesList extends React.Component {
     onNeedMoreLines: PropTypes.func.isRequired,
   }
 
+  state = {
+    sortOrder: ['name', 'activeFrom', 'activeTo'],
+    sortDirection: ['asc', 'desc'],
+  };
+
   columnWidths = {
     name: 250,
     provider: 150,
     coverage: 250,
   }
 
+  sortMap = {
+    name: line => line.resource.name,
```

Comment:
I think pulling in the `name` in the sortMap from `line.resource.name` will be an issue because the name might not always live under that object. For example: For a detached line we may not have a name and only have a description. This will probably break the UI. So firstly `line.resource.name` needs to be optional chained `line.resource?.name`  and then also all the different cases like `detached` line, `eholdings` line and `internal` line need to be tested for sorting.

Reply:
I made the changes